opal/runtime/opal_progress.c

```diff
@@ -66,6 +67,10 @@ static size_t callbacks_lp_size = 0;
 
 /* do we want to call sched_yield() if nothing happened */
 bool opal_progress_yield_when_idle = false;
+/* do we want to poll() if nothing happened for a while */
+uint32_t opal_progress_poll_threshold = 1000;
```

Comment:
I'm guessing you chose this number fairly arbitrarily.

Any idea how it impacts real application performance?

Put differently: any idea how long it takes to go through 1,000 cycles of the progress engine in an "idle" MPI process?

My *guess* is that 1,000 is going to be too low (i.e., we'll go into `poll()` mode too quickly, which could impact latency-sensitive applications).